tests/suites/test_suite_arc4.function

```diff
@@ -7,6 +7,26 @@
  * END_DEPENDENCIES
  */
 
+/* BEGIN_CASE depends_on:MBEDTLS_CHECK_PARAMS */
+void mbedtls_arc4_invalid_param( )
+{
+    mbedtls_arc4_context ctx;
+    unsigned char key[128] = { 0 };
+
+    /* Check that parameter validation detect NULL parameters. */
+    TEST_INVALID_PARAM( mbedtls_arc4_init( NULL ) );
+    TEST_INVALID_PARAM( mbedtls_arc4_setup( NULL, key, sizeof( key ) ) );
+    TEST_INVALID_PARAM( mbedtls_arc4_setup( &ctx, NULL, 42 ) );
+
+    /* Check that parameter validation detects keys of invalid length. */
+    TEST_INVALID_PARAM( mbedtls_arc4_setup( &ctx, key, 4 ) );
+    TEST_INVALID_PARAM( mbedtls_arc4_setup( &ctx, key, 257 ) );
+
```

Comment:
Please also test that `NULL` is accepted by the `free()` function.

Reply:
If we want to test the behaviour of `free()`, I think it should be done in a separate test, because it's not affected by the presence of parameter validation.

Reply:
I think testing it somewhere is better than not testing it at all, but indeed testing regardless of the value of `MBEDTLS_CHECK_PARAMS` is even better. I'm updating #1643 to make `TEST_VALID_PARAM()` available in all builds.